tests/scripts/check-python-files.sh

```diff
@@ -62,6 +62,7 @@ $PYTHON -m pylint scripts/mbedtls_dev/*.py scripts/*.py tests/scripts/*.py || {
 
 echo
 echo 'Running mypy ...'
+export MYPYPATH="../../scripts"
```

Comment:
I'd prefer not to need that, and we didn't need it before. Why do we need it now?

In any case, this looks wrong: for `scripts/foo.py`, this tells mypy to look outside the mbedtls tree?

Reply:
> Why do we need it now?

mypy was complaining for not finding `asymmetric_key_data`. I think this is because `generate_test_keys.py` is in `[root]/tests/scripts`, while `asymmetric_key_data` is in `[root]/scripts/mbedtls_dev`.
Even the import on top of the file looks quite ugly:
```
SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__)) + "/"
sys.path.append(SCRIPT_DIR + "../../scripts/")
from mbedtls_dev.asymmetric_key_data import ASYMMETRIC_KEY_DATA
```

but I didn't find any better solution. The only option I had in mind was to move `generate_test_keys` to `[root]/scripts/mbedtls_dev`, but that didn't look as the right place for a test script. 
Any idea is absolutely welcome :)

Reply:
I applied the [solution](https://github.com/Mbed-TLS/mbedtls/pull/8986#discussion_r1561519572) you proposed in #8986. Thanks again!